scripts/install.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash +x
+#
+# This file is part of Moonfire NVR, a security camera network video recorder.
+# Copyright (C) 2016-17 Scott Lamb <slamb@slamb.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# In addition, as a special exception, the copyright holders give
+# permission to link the code of portions of this program with the
+# OpenSSL library under certain conditions as described in each
+# individual source file, and distribute linked combinations including
+# the two.
+#
+# You must obey the GNU General Public License in all respects for all
+# of the code used other than OpenSSL. If you modify file(s) with this
+# exception, you may extend this exception to your version of the
+# file(s), but you are not obligated to do so. If you do not wish to do
+# so, delete this exception statement from your version. If you delete
+# this exception statement from all source files in the program, then
+# also delete it here.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+. `dirname ${BASH_SOURCE[0]}`/script-functions.sh
+
+# Determine directory path of this script
+#
+initEnvironmentVars
+
+# Process command line options
+#
+while getopts ":s" opt; do
+	case $opt in
+	  s)    NO_SERVICE=1
+		;;
+	  :)
+		echo "Option -$OPTARG requires an argument." >&2
+		exit 1
+		;;
+	  \?)
+		echo "Invalid option: -$OPTARG" >&2
+		exit 1
+		;;
+	esac
+done
+
+
+sudo install -m 755 target/release/moonfire-nvr ${SERVICE_BIN}
+if [ -x "${SERVICE_BIN}" ]; then
+	echo "Server Binary installed..."; echo
+else
+	echo "Server build failed to install..."; echo
+	exit 1
+fi
+if [ ! -d "${LIB_DIR}" ]; then
+	sudo mkdir "${LIB_DIR}"
+fi
+sudo cp -R ui-dist "${LIB_DIR}/ui"
+if [ -d "ui-dist" ] && [ -d "${LIB_DIR/ui}" ]; then
+	echo "Server UI installed..."; echo
+else
+	echo "Server UI failed to build or install..."; echo
+	exit 1
+fi
+
+if [ "${NO_SERVICE:-0}" != 0 ]; then
+	echo "Not configuring systemctl service. Done"; echo
+	exit 0
+fi
+
+# Make sure user was created
+#
+if ! userExists "${NVR_USER}"; then
+	echo "NVR_USER=${NVR_USER} was not created. Do so manually"
+	echo "or run the setup script."
+	exit 1
+fi
+
+
+# Prepare service files for socket when using priviliged port
```

Comment:
Could we skip this part for now, and the suggestion in the guide to use port 80? Having it here suggests that it will work. From your link, it sounds not terribly hard to implement but it's not quite at the top of my list. (If you'd like to do it, I'd be happy to review/advise though.)

If anything we should just have if [ $NVR_PORT -lt 1024 ]; then echo "error: unsupported..." 2>&1; ...; fi